activerecord/lib/active_record/fixtures.rb

```diff
@@ -578,12 +594,16 @@ def initialize(connection, name, class_name, path, config = ActiveRecord::Base)
       @name     = name
       @path     = path
       @config   = config
-      @model_class = nil
 
-      if class_name.is_a?(Class) # TODO: Should be an AR::Base type class, or any?
-        @model_class = class_name
-      else
-        @model_class = class_name.safe_constantize if class_name
+      self.model_class = class_name
+
+      @fixtures = read_fixture_files(path)
+
+      # Sets the custom model class to all loaded fixtures.
+      fixture_settings = @fixtures.delete('_fixture')
+      if fixture_settings && (custom_model_class = fixture_settings['model_class'])
+        self.model_class = custom_model_class
+        @fixtures.values.each { |fixture| fixture.model_class = self.model_class }
```

Comment:
I don't like that we create `ActiveRecord::Fixture` objects with the "wrong" `model_class` and then re-assign here. Can we find a way that moves knowledge about `_fixture:model_class` into `Fixture::File` so that we can create `Fixture` objects with the right `model_class` from the get go?